docker/docker/src/main/java/org/arquillian/cube/docker/impl/client/CubeDockerConfiguration.java

```diff
@@ -307,4 +315,24 @@ private static String resolveFilename(URI uri) {
 
         return content.toString();
     }
+
+    public static String resolveUrl(String machineVersion) {
+        validation(machineVersion, DOCKER_MACHINE_VERSION);
+        return "https://github.com/docker/machine/releases/download/v" + machineVersion + "/" + DockerMachineDistro.resolveDistro();
+    }
+
+    public static String resolveMachinePath(String machineCustomPath, String machineVersion) {
+        validation(machineVersion, DOCKER_MACHINE_VERSION);
+        if (StringUtils.isBlank(machineCustomPath)) {
+            machineCustomPath = DOCKER_MACHINE_ARQUILLIAN_PATH;
+        }
+        return HomeResolverUtil.resolveHomeDirectoryChar(machineCustomPath + "/" + machineVersion + "/" + DOCKER_MACHINE_COMMAND);
+    }
+
+    private static void validation(String attribute, String attributeName) {
+        if (StringUtils.isBlank(attribute)) {
+            throw new IllegalArgumentException("Attribute '" + attributeName +"' is missing.");
```

Comment:
Would be nice to be told why this attribute is important and why i need to specify it. e.g.: Could not attempt docker-machine automatic install, no attributeName defined.